src/services/codefixes/fixAddMissingMember.ts

```diff
@@ -31,8 +31,9 @@ namespace ts.codefix {
                 const changes = textChanges.ChangeTracker.with(context, t => addJsxAttributes(t, context, info));
                 return [createCodeFixAction(fixMissingAttributes, changes, Diagnostics.Add_missing_attributes, fixMissingAttributes, Diagnostics.Add_all_missing_attributes)];
             }
-            if (info.kind === InfoKind.Function) {
-                const changes = textChanges.ChangeTracker.with(context, t => addFunctionDeclaration(t, context, info));
+            if (info.kind === InfoKind.Function || info.kind === InfoKind.Signature) {
+                const changes = textChanges.ChangeTracker.with(context, t =>
+                    info.kind === InfoKind.Signature ? addFunctionDeclarationFromSignature(t, context, info) : addFunctionDeclaration(t, context, info));

```

Comment:
I would move the check for InfoKind.Signature inside addFunctionDeclaration since it would simplify the code. 

From reading addFunctionDeclaration I can't quite tell if you're better off with two different functions or not -- maybe the code could be unified.

Reply:
Oke, moved